cmake/AzAutoGen.py

```diff
@@ -37,7 +37,7 @@ def __init__(self, targetName, cacheDir, outputDir, projectDir, inputFiles, expa
         self.pythonPaths = pythonPaths
 
 def SanitizeTargetName(targetName):
-    return targetName.replace('.', '').replace('::', '')
+    return ''.join(ch for ch in targetName if ch.isalnum())
```

Comment:
Wouldn't it be safer to replace non alphanumeric characters with an underscore?

Reply:
As original we are using empty string to replace informal character, using underscore will require changes in other consumer places. (Which we can address it separately if user find out that is an issue, for example, if user gives project name like 1234)